src/ucp/wireup/wireup_cm.c

```diff
@@ -496,6 +552,19 @@ static unsigned ucp_cm_client_uct_connect_progress(void *arg)
         goto try_fallback;
     }
 
+    status = ucp_cm_ep_config_init_lanes(ep, &key, ep_init_flags_prio[i]);
+    if (status != UCS_OK) {
+        goto err;
+    }
+
+    ucp_ep_get_tl_bitmap(&key, &tl_bitmap);
+    status = ucp_cm_ep_priv_data_pack(ep, &tl_bitmap,
+             context->config.ext.sa_client_min_hdr_version, &priv_data,
+             &priv_data_length, ep_pack_flags);
+    if (status != UCS_OK) {
+        goto err;
+    }
```

Comment:
From code implementation perspective, this change can be reverted. However, it seems it's not proper to do this.
The ```for-loop``` is mainly to get right ```ep_config_key``` and ```ep_init_flags```.  It may make the ```for-loop``` complex once reverting this change.